spec/controllers/alaveteli_pro/subscriptions_controller_spec.rb

```diff
@@ -0,0 +1,284 @@
+        let(:token) { stripe_helper.generate_card_token }
+
+        before do
+          error = Stripe::AuthenticationError.new
+          StripeMock.prepare_error(error, :create_subscription)
+          post :create, 'stripeToken' => token,
+                        'stripeTokenType' => 'card',
+                        'stripeEmail' => user.email,
+                        'plan_id' => 'pro'
+        end
+
+        it 'sends an exception email' do
+          mail = ActionMailer::Base.deliveries.first
+          expect(mail.subject).to match(/Stripe::AuthenticationError/)
+        end
+
+        it 'renders an error message' do
+          expect(flash[:error]).to match(/There was a problem/)
+        end
+
+        it 'redirects to the plan page' do
+          expect(response).to redirect_to(plan_path('pro'))
+        end
+
+      end
+
+      context 'when we cannot connect to Stripe' do
+        let(:token) { stripe_helper.generate_card_token }
+
+        before do
+          error = Stripe::APIConnectionError.new
+          StripeMock.prepare_error(error, :create_subscription)
+          post :create, 'stripeToken' => token,
+                        'stripeTokenType' => 'card',
+                        'stripeEmail' => user.email,
+                        'plan_id' => 'pro'
+        end
+
+        it 'sends an exception email' do
+          mail = ActionMailer::Base.deliveries.first
+          expect(mail.subject).to match(/Stripe::APIConnectionError/)
+        end
+
+        it 'renders an error message' do
+          expect(flash[:error]).to match(/There was a problem/)
+        end
+
+        it 'redirects to the plan page' do
+          expect(response).to redirect_to(plan_path('pro'))
+        end
+
+      end
+
+      context 'when Stripe returns a generic error' do
+        let(:token) { stripe_helper.generate_card_token }
+
+        before do
+          error = Stripe::StripeError.new
+          StripeMock.prepare_error(error, :create_subscription)
+          post :create, 'stripeToken' => token,
+                        'stripeTokenType' => 'card',
+                        'stripeEmail' => user.email,
+                        'plan_id' => 'pro'
+        end
+
+        it 'sends an exception email' do
+          mail = ActionMailer::Base.deliveries.first
+          expect(mail.subject).to match(/Stripe::StripeError/)
+        end
+
+        it 'renders an error message' do
+          expect(flash[:error]).to match(/There was a problem/)
+        end
+
+        it 'redirects to the plan page' do
+          expect(response).to redirect_to(plan_path('pro'))
+        end
+
+      end
+
+      context 'when invalid params are submitted' do
+
+        it 'redirects to the plan page if there is a plan' do
+          post :create, :plan_id => 'pro'
+          expect(response).to redirect_to(plan_path('pro'))
+        end
+
+        pending 'redirects to the pricing page if there is no plan' do
```

Comment:
this syntax appears to cause 2.0.0 to throw actual RSpec errors - it doesn't like the undefined path variable - instead of treating the block as pending :(

removing the do block and supplying the code as a comment instead seems to work